src/v/security/audit/audit_log_manager.cc

```diff
@@ -501,6 +527,7 @@ ss::future<> audit_log_manager::start() {
 }
 
 ss::future<> audit_log_manager::stop() {
+    _probe.reset(nullptr);
```

Comment:
Is this safe? Since the gate is still open at this point, between the next scheduling point it may be the case that an executing future attempts to use a reference to a just-deleted resource. I think omitting this, or moving this line until after the gate is closed is important

Reply:
Yeah, great catch.

Reply:
Moved to a safe place but left both of the `reset`s in, mostly to be explicit about intent. "Nuke these metrics when the enclosing thing is switched off"